code/player/Player.cs

```diff
@@ -189,30 +193,39 @@ private void TickPlayerDropWeapon()
 
         private void TickAttemptInspectPlayerCorpse()
         {
-            if (IsServer)
+            using (Prediction.Off())
             {
-                using (Prediction.Off())
-                {
-                    To client = To.Single(this);
-                    PlayerCorpse playerCorpse = IsLookingAtPlayerCorpse();
+                To client = To.Single(this);
+                PlayerCorpse playerCorpse = IsLookingAtPlayerCorpse();
 
-                    if (playerCorpse != null)
+                if (playerCorpse != null)
+                {
+                    if (inspectingPlayerCorpse != playerCorpse)
                     {
-                        if (Input.Down(InputButton.Use) && !playerCorpse.IsIdentified)
-                        {
-                            playerCorpse.IsIdentified = true;
-
-                            ClientConfirmPlayer(this, playerCorpse.Player, playerCorpse.Player.Role.Name);
-                        }
+                        inspectingPlayerCorpse = playerCorpse;
 
                         // Send the request to the player looking at the player corpse.
                         // https://wiki.facepunch.com/sbox/RPCs#targetingplayers
                         ClientOpenInspectMenu(client, playerCorpse.Player, playerCorpse.IsIdentified);
+                    }
+
+                    if (!playerCorpse.IsIdentified && Input.Down(InputButton.Use))
+                    {
+                        playerCorpse.IsIdentified = true;
 
-                        return;
+                        ClientConfirmPlayer(this, playerCorpse.Player, playerCorpse.Player.Role.Name);
+
+                        ClientOpenInspectMenu(client, playerCorpse.Player, playerCorpse.IsIdentified);
                     }
 
+                    return;
+                }
+
+                if (inspectingPlayerCorpse != null)
+                {
                     ClientCloseInspectMenu(client);
+
+                    inspectingPlayerCorpse = null;
```

Comment:
Is there any cases where the inspect menu fails to close?

Reply:
not in my tests, but feel free trying some edge-cases :D